app/src/main/java/com/example/studentarena/fragments/ComposeFragment.java

```diff
@@ -1,66 +1,169 @@
-    private String mParam1;
-    private String mParam2;
+                if (photoFile == null || ivPostImage.getDrawable() == null){
+                    Toast.makeText(getContext(), "There is no image!", Toast.LENGTH_SHORT).show();
+                    return;
+                }
+                ParseUser currentUser = ParseUser.getCurrentUser();
+                savePost (description, contact, title, price, currentUser, photoFile);
+            }
+        });
+    }
 
-    public ComposeFragment() {
-        // Required empty public constructor
+    private void savePost(String description, String contact, String title, String price, ParseUser currentUser, File photoFile) {
+        Post post = new Post();
+        post.setDescription(description);
+        post.setContact(contact);
+        post.setPrice(price);
+        post.setTitle(title);
+        post.setImage(new ParseFile(photoFile));
+        post.setUser((User) currentUser);
+        post.saveInBackground(new SaveCallback() {
+            @Override
+            public void done(ParseException e) {
+                if (e != null) {
+                    Log.e(TAG, "Error while saving", e);
+                    Toast.makeText(getContext(), "Error while saving!", Toast.LENGTH_SHORT).show();
+                } else {
+                    Log.i(TAG, "Post save was successful!!");
+                    getActivity().getSupportFragmentManager()
+                            .beginTransaction()
+                            .replace(R.id.flContainer, new FeedFragment())
+                            .commit();
+//                    etDescription.setText("");
+//                    ivPostImage.setImageResource(0);
+                }
+            }
+        });
     }
 
-    /**
-     * Use this factory method to create a new instance of
-     * this fragment using the provided parameters.
-     *
-     * @param param1 Parameter 1.
-     * @param param2 Parameter 2.
-     * @return A new instance of fragment ComposeFragment.
-     */
-    // TODO: Rename and change types and number of parameters
-    public static ComposeFragment newInstance(String param1, String param2) {
-        ComposeFragment fragment = new ComposeFragment();
-        Bundle args = new Bundle();
-        args.putString(ARG_PARAM1, param1);
-        args.putString(ARG_PARAM2, param2);
-        fragment.setArguments(args);
-        return fragment;
+    private void launchCamera() {
+        Intent intent = new Intent( MediaStore.ACTION_IMAGE_CAPTURE);
+        // Create a File reference for future access
+        photoFile = getPhotoFileUri(photoFileName);
+        Uri fileProvider = FileProvider.getUriForFile(getContext(), "com.codepath.fileprovider", photoFile);
+        intent.putExtra(MediaStore.EXTRA_OUTPUT, fileProvider);
+        // If you call startActivityForResult() using an intent that no app can handle, your app will crash.
+        // So as long as the result is not null, it's safe to use the intent.
```

Comment:
great example of explaining something you're doing now in case you forget why you did it in 6 months :)